Production/src/Selectors.cc

```diff
@@ -177,5 +179,123 @@ TauJetSelector::Result TauJetTag::Select(const edm::Event& event, const std::deq
+		    continue;
+	    if(!selected || selected->jet->pt() < jet.pt()) {
+		    selected = &tauJet;
+		    selectedTauJets = { selected };
+            }
+	}
+	return Result(selectedTauJets, tagObject);
+    }
+    //	
+    else if(ref_electron) { // Fill tag with ele if ele found rather than muon
+	tagObject->kind = tau_analysis::selectors::TagObject::Kind::Electron; //new tag object member
+        tagObject->p4 = ref_electron->polarP4();
+        tagObject->charge = ref_electron->charge();
+        tagObject->id = ref_electron->electronID("mvaEleID-Fall17-noIso-V2-wp90"); //Not sure what to put here 
+        tagObject->isolation = PFRelIsolation(*ref_electron,rho);
+        tagObject->has_extramuon = hasExtraMuon(muons, ref_muon, primaryVertex);
+        tagObject->has_extraelectron = hasExtraElectron(electrons, rho);
+        tagObject->has_dimuon = hasExtraDimuon(muons, primaryVertex);	
+	const TauJet* selected = nullptr;
+        for(const TauJet& tauJet : tauJets) { //Look for highest pT jet
+            if(!tauJet.jet) continue;
+            const pat::Jet& jet = *tauJet.jet;
+            if(!(std::abs(jet.eta()) < 2.7 && reco::deltaR(jet.polarP4(), ref_electron->polarP4()) > 0.8))
+                    continue;
+            if(!selected || selected->jet->pt() < jet.pt()) {
+                    selected = &tauJet;
+		    selectedTauJets = { selected };
+            }
+        }
+        return Result(selectedTauJets, tagObject);
+    }
+    //
+    else if(met.pt() > 180) { // If no mu or ele tag, then look for event with MET>180GeV as third option
+	const TauJet* selected = nullptr;
+        for(const TauJet& tauJet : tauJets) { //Look for highest pT jet
+            if(!tauJet.jet) continue;
+            const pat::Jet& jet = *tauJet.jet;
+            if(!(std::abs(jet.eta()) < 2.8))
+                    continue;
+            if(!selected || selected->jet->pt() < jet.pt()) {
+                    selected = &tauJet;
+                    selectedTauJets = { selected };
+            }
+        }
+        return Result(selectedTauJets, tagObject);
+    }	
+    //
+    else { // No mu or ele tag nor high MET 
+        for(const TauJet& tauJet : tauJets) {
+            if(!(tauJet.genLepton && tauJet.genJet) && !(std::abs(tauJet.genLepton->visibleP4().eta()) < 2.8 || std::abs(tauJet.genJet->eta()) < 2.8)) continue;
+	    selectedTauJets.push_back(&tauJet); //Fill with jets passing selection
+	}
+	if(selectedTauJets.size() <= 2)
+	    return Result(selectedTauJets, nullptr); //Send result if vector size is already <= 2
+	else { // Otherwise order by genLepton properties
+            std::vector<const TauJet*> newSelectedTauJets;
+	    std::vector<int> genLeptonKind{5,3,4,2,1,6}; // Ordered by lepton kind priority, couldn't come up with a more efficient idea for now...
+	    std::sort(selectedTauJets.begin(), selectedTauJets.end(), [](auto &i, auto &j){return i->genLepton->visibleP4().pt() < j->genLepton->visibleP4().pt();});
```

Comment:
- no need to treat size < 2 separately. After the vector is sorted it is enough to call `selectedTauJets.resize(std::min(selectedTauJets.size(), 2))`

I propose following modification: don't include kind priority also for di-tau case - save two jets with the highest pt. However, for all cases (with tag/MET/ditau) in addition to the selected tauJets add all tauJets that have genJepton.kind() == 5 (i.e. true taus). This would allow us
- unbiased estimation of tau reco efficiency
- unbiased selection of the jet pair for di-tau case that is the same for data&mc => also ditau can be used for domain adaptation.

Reply:
So if I understand correctly we would have :

- 1 reco jet with highest pT for tag/MET as we did before + all genLeptons with kind == 5 
- 2 genLeptons with highest pT for ditau regardless of their kind + all genLeptons with kind == 5 instead of two genLeptons sorted by kind priority